ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+        {
+            free_function(pszBuffer);
+            fclose(pfMMSRS);
+            MMCPLError(CE_Failure, CPLE_NotSupported,
+                       "Wrong format in data\\MM_m_idofic.csv.\n");
+            return 1;
+        }
+
+        psidgeodes = strstr(id_geodes + 1, ";");
+        if (!psidgeodes)
+        {
+            free_function(pszBuffer);
+            fclose(pfMMSRS);
+            MMCPLError(CE_Failure, CPLE_NotSupported,
+                       "Wrong format in data\\MM_m_idofic.csv.\n");
+            return 1;
+        }
+
+        id_geodes[(ptrdiff_t)psidgeodes - (ptrdiff_t)id_geodes] = '\0';
+        psidgeodes = pszBuffer;
+        psidgeodes[(ptrdiff_t)id_geodes - (ptrdiff_t)psidgeodes] = '\0';
+        id_geodes++;
+
+        if (direction == EPSG_FROM_MMSRS)
+        {
+            // I have pMMSRS and I want pSRS
+            if (strcmp(pMMSRS_or_pSRS, id_geodes))
+                continue;
+
+            epsg = strstr(psidgeodes, "EPSG:");
+            nLong = strlen("EPSG:");
+            if (epsg && !strncmp(epsg, psidgeodes, nLong))
+            {
+                if (epsg[nLong] != '\0')
+                {
+                    strcpy(szResult, epsg + nLong);
+                    free_function(pszBuffer);
+                    fclose(pfMMSRS);
+                    return 0;  // found
+                }
+                else
+                {
+                    free_function(pszBuffer);
+                    fclose(pfMMSRS);
+                    *szResult = '\0';
+                    return 1;  // not found
+                }
+            }
+        }
+        else
+        {
+            // I have pSRS and I want pMMSRS
+            epsg = strstr(psidgeodes, "EPSG:");
+            nLong = strlen("EPSG:");
+            if (epsg && !strncmp(epsg, psidgeodes, nLong))
+            {
+                if (epsg[nLong] != '\0')
+                {
+                    if (!strcmp(pMMSRS_or_pSRS, epsg + nLong))
+                    {
+                        strcpy(szResult, id_geodes);
+                        free_function(pszBuffer);
+                        fclose(pfMMSRS);
+                        return 0;  // found
+                    }
+                }
+            }
+        }
+    }
+
+    free_function(pszBuffer);
+    fclose(pfMMSRS);
+    return 1;  // not found
+}
+
+#define LineReturn "\r\n"
+
+// Generates an idientifier that REL 4 MiraMon metadata needs.
+static char *MMGenerateFileIdentifierFromMetadataFileName(char *pMMFN)
+{
+    static char aCharRand[8],
+        aCharset[] =
+            "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+    int i, len_charset;
+    static char aFileIdentifier[MM_MAX_LEN_LAYER_IDENTIFIER];
+
+    memset(aFileIdentifier, '\0', MM_MAX_LEN_LAYER_IDENTIFIER);
+
+    aCharRand[0] = '_';
+    len_charset = (int)strlen(aCharset);
+    for (i = 1; i < 7; i++)
+        aCharRand[i] = aCharset[rand() % (len_charset - 1)];
```

Comment:
rand() is not so random. It might be wise to call srand(time(NULL)) somewhere before. possibly at Create() or CreateLayer() time

Reply:
Like you put here it gives a warning:
warning C4244: 'argument': conversion from 'time_t' to 'unsigned int', possible loss of data

I need to cast.
srand((unsigned int)time(NULL));